### ISSUE TITLE
Out of Memory after training a few epochs

### ISSUE THREAD

**Issue by waylonli** [mistralai/mistral-src#20]
The code I'm using is in file "one_file_ref".
I was trying to apply Mistral Transformer on other non-text tubular data. I initialised "positions" as torch.arange(1, num_of_most_instances) where "num_of_most_instances" is equivalent to the number of tokens in the longest sequence.
However, I have observed that each time I called loss.backward() and enter the next batch, there would be 30mb of gpu memory which could not be released. Thus, after 1000 steps it took 30gb of gpu memory.
I have comment the codes related to self.cache and the issue was resolved. Also I found that it always entered line 131 and never went into "else". 
Is there any mistake of ny usage of "positions"? Though the issue does not happen again after I comment out the self.cache stuff, I'm wondering if that will affect the attention mechanism.

**Comment by juliendenize** [comment]
Hi! We're archiving this repository as it's no longer actively maintained, and therefore this issue is being closed.
